src/hotspot/cpu/aarch64/abstractInterpreter_aarch64.cpp

```diff
@@ -165,6 +165,19 @@ void AbstractInterpreter::layout_activation(Method* method,
     popframe_extra_args;
   interpreter_frame->interpreter_frame_set_last_sp(esp);
 
+  // We have to add extra reserved slots to max_stack. There are 3 users of the extra slots,
+  // none of which are at the same time, so we just need to make sure there is enough room
+  // for the biggest user:
+  //   -reserved slot for exception handler
+  //   -reserved slots for JSR292. Method::extra_stack_entries() is the size.
+  //   -reserved slots for TraceBytecodes
+  int max_stack = method->constMethod()->max_stack() + MAX2(3, Method::extra_stack_entries());
+  intptr_t* extended_sp = (intptr_t*) monbot  -
+    (max_stack * Interpreter::stackElementWords) -
```

Comment:
You are correctly multiplying by Interpreter::stackElementWords here. However, I noticed that at line 93 the multiply has been omitted
```
  int size = overhead +
         (callee_locals - callee_params) +
         monitors * frame::interpreter_frame_monitor_size() +
```
It should really follow x86 and include the multiply:
```
  int size = overhead +
         (callee_locals - callee_params) * Interpreter::stackElementWords +
         monitors * frame::interpreter_frame_monitor_size() +
```
We only get away with this because Interpreter::stackElementWords is 1.

Perhaps you could correct that as part of this patch?

Reply:
OK, done. Ta.